cinemy/cinema/views.py

```diff
@@ -114,3 +114,11 @@ def select_seats(request, pk):
                 [user.email],
             )
         return redirect("front_page")
+
+
+@login_required()
+@ratelimit(key="ip", rate="30/m", block=True)
+def movie_detail(request, pk):
+    if request.method == "GET":
+        movie = get_object_or_404(Movie, id=pk)
+        return render(request, "cinema/movie_details.html", {"movie": movie})
```

Comment:
See if you can use this here
https://docs.djangoproject.com/en/4.0/ref/class-based-views/generic-display/#django.views.generic.detail.DetailView

Example:
class DetailView(generic.DetailView):
    model = Question
    template_name = 'polls/details.html'

You might even skip the template_name assignment if the naming convention is used.